tests/test_utils.py

```diff
@@ -138,3 +139,24 @@ def test_output_structs(
 
     else:
         assert not output_file.exists()
+
+
+@pytest.mark.parametrize(
+    "dicts",
+    [
+        [None, {"a": 1}, {}, {"b": 2, "a": 11}, None],
+        (None, {"a": 1}, {}, {"b": 2, "a": 11}, None),
+    ],
+)
+def test_none_to_dict(dicts):
+    """Test none_to_dict removes Nones from sequence, and preserves dictionaries."""
+    dicts = list(none_to_dict(dicts))
+    for dictionary in dicts:
+        assert dictionary is not None
+
+    assert dicts[0] == {}
+    assert dicts[1]["a"] == 1
+    assert dicts[2] == {}
+    assert dicts[3]["b"] == 2
+    assert dicts[3]["a"] == 11
+    assert dicts[4] == {}
```

Comment:
```suggestion
@pytest.mark.parametrize(
    "dicts_in",
    [
        [None, {"a": 1}, {}, {"b": 2, "a": 11}, None],
        (None, {"a": 1}, {}, {"b": 2, "a": 11}, None),
    ],
)
def test_none_to_dict(dicts_in):
    """Test none_to_dict removes Nones from sequence, and preserves dictionaries."""
    dicts = list(none_to_dict(dicts))
    for dictionary in dicts:
        assert dictionary is not None

    assert dicts[0] == {}
    assert dicts[1] == dicts_in[1]
    assert dicts[2] == dicts_in[2]
    assert dicts[3] == dicts_in[3]
    assert dicts[4] == {}
```

Reply:
Is this supposed to be `dicts = list(none_to_dict(dicts_in))`?

What's the benefit/aim generally?

Reply:
Yes. Just to avoid having the manual checks of every element. Just simplifying the logical info.

Reply:
Fair enough, I sort of like having it closer to the way we use it, but it's the same thing really, so yeah probably neater